### PR TITLE
Hook and heartbeat updates

### PR DESCRIPTION
**Pull request by meatballhat**
to work moar better with cyclist and terraform config

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -198,6 +191,48 @@ func (i *CLI) Run() {
...
+	}).Error("start hook failed")
+}
+
+func (i *CLI) handleStopHook() {
+	hookValue := i.c.String("stop-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("stop_hook", hookValue).Info("adding deferred execution")
```
Comment (henrikhodne) [inline cli.go]: This line doesn't seem correct anymore. It used to be called before we did the `defer`, but now it's run "inside" of the `defer`-ed function.

File: cli.go [hunk 2]
```diff
@@ -198,6 +191,48 @@ func (i *CLI) Run() {
...
+
+	parts := stringSplitSpace(hookValue)
+	outErr, err := exec.Command(parts[0], parts[1:]...).CombinedOutput()
```
Comment (henrikhodne) [inline cli.go]: Are there any multi-word parameters to the commands we use (for example `/bin/sh -c "foo bar"`)? If so, this won't work, but if everything is single-word it should work fine.

Comment (meatballhat) [inline cli.go]: I'm not planning to add anything multi-word, but yeap it's definitely not foolproof.

File: cli.go [hunk 3]
```diff
@@ -198,6 +190,68 @@ func (i *CLI) Run() {
...
+	go i.heartbeatHandler(hbURL, strings.TrimSpace(hbTok))
+}
+
+func (i *CLI) handleStartHook() {
+	hookValue := i.c.String("start-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("start_hook", hookValue).Info("running")
```
Comment (henrikhodne) [inline cli.go]: Would a better message be "running hook" or "running start hook"? I'm thinking if we want to search for these log messages, searching for `msg="running start hook"` might be nicer than `msg="running" start_hook=`?

File: cli.go [hunk 4]
```diff
@@ -198,6 +190,68 @@ func (i *CLI) Run() {
...
+	}).Error("start hook failed")
+}
+
+func (i *CLI) handleStopHook() {
+	hookValue := i.c.String("stop-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("stop_hook", hookValue).Info("running")
```
Comment (henrikhodne) [inline cli.go]: Similarly, should this be "running hook" or "running stop hook" instead?

**Comment by henrikhodne** [comment]
LGTM now :+1: Left two more questions above, but I'll leave it up to you if you want to change that or not.

**Pull request merged**